### PR TITLE
d3d11: retry device creation without debug, if SDK is not available

### LINKED ISSUE #11512: [vo/gpu-next/d3d11] Failed to create Direct3D 11 device
### Important Information

Provide following Information:
- mpv version
```
mpv 0.35.0-339-gd1d0b1a8 Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
 built on Tue Mar 28 12:41:04 2023
libplacebo version: v5.264.0-85-g0f36b01
FFmpeg version: N-110105-g9bf1848ac
FFmpeg library versions:
   libavutil       58.5.100
   libavcodec      60.7.100
   libavformat     60.4.100
   libswscale      7.2.100
   libavfilter     9.5.100
   libswresample   4.11.100
```
- Windows Version
```
Edition	Windows 10 Pro
Version	22H2
Installed on	‎19/‎03/‎2023
OS build	19045.2728
Experience	Windows Feature Experience Pack 120.2212.4190.0
```
- Source of the mpv binary https://github.com/shinchiro/mpv-winbuild-cmake/releases
- If known which version of mpv introduced the problem: No
- Possible screenshot or video of visual glitches: No

### Reproduction steps
```mpv --no-config --fs --log-file=Example1-0005.txt --gpu-debug --vo=gpu-next Example1-0005.m2ts```

### Expected behavior
mpv able to use d3d11

### Actual behavior
```
[   0.121][v][vo/gpu-next] Probing for best GPU context.
[   0.121][v][vo/gpu-next/d3d11] Initializing GPU context 'd3d11'
[   0.689][d][vo/gpu-next/d3d11] Failed to create 12_0+ device, trying 11_1
[   1.249][d][vo/gpu-next/d3d11] Failed to create 11_1+ device, trying 11_0
[   1.810][d][vo/gpu-next/d3d11] Failed to create hardware device, trying WARP
[   1.812][d][vo/gpu-next/d3d11] Failed to create 12_0+ device, trying 11_1
[   1.813][d][vo/gpu-next/d3d11] Failed to create 11_1+ device, trying 11_0
[   1.814][f][vo/gpu-next/d3d11] Failed to create Direct3D 11 device: The application requested an operation that depends on an SDK component that is missing or mismatched. (0x887a002d)
[   1.814][v][vo/gpu-next/opengl] Initializing GPU context 'angle'
[   1.814][v][vo/gpu-next] Failed to load LIBEGL.DLL
[   1.815][v][vo/gpu-next/opengl] Initializing GPU context 'win'
[   1.880][v][vo/gpu-next] GL_VERSION='4.6.0 NVIDIA 531.41'
[   1.880][v][vo/gpu-next] Detected desktop OpenGL 4.6.
[   1.880][v][vo/gpu-next] GL_VENDOR='NVIDIA Corporation'
```

### Log file
[Example1-0005.txt](https://github.com/mpv-player/mpv/files/11089805/Example1-0005.txt)

### PR DESCRIPTION
**Pull request by kasper93**
Fixes #11512

### CODE REVIEW

File: video/out/gpu/d3d11_helpers.c [hunk 1]
```diff
@@ -510,7 +511,14 @@ bool mp_d3d11_create_present_device(struct mp_log *log,
         max_fl = max_fl ? max_fl : D3D_FEATURE_LEVEL_11_0;
         min_fl = min_fl ? min_fl : D3D_FEATURE_LEVEL_9_1;
 
-        hr = create_device(log, adapter, warp, opts->debug, max_fl, min_fl, &dev);
+        hr = create_device(log, adapter, warp, debug, max_fl, min_fl, &dev);
+
+        // Retry without debug, if SDK is not available
+        if (debug && hr == DXGI_ERROR_SDK_COMPONENT_MISSING) {
+            debug = false;
```
Comment (haasn) [inline video/out/gpu/d3d11_helpers.c:518]: Add an `MP_WARN` explaining that debugging was disabled due to debug SDK not being installed

**Review by haasn — DISMISSED** [review]

File: video/out/gpu/d3d11_helpers.c (hunk 1 above)
Comment (kasper93) [reply video/out/gpu/d3d11_helpers.c:518]: Heh, good idea. Was just doing that. I though create_device will print something, but here there is not fancy `D3D` macro like in placebo.

Comment (kasper93) [reply video/out/gpu/d3d11_helpers.c:519]: Are you ok with 

```
[vo/gpu-next/d3d11] gpu-debug disabled due to error: The application requested an operation that depends on an SDK component that is missing or mismatched. (0x887a002d)
```

Or should I make it shorter, like

```
Debug layers not available, removing debug flag
```

Comment (haasn) [reply video/out/gpu/d3d11_helpers.c:519]: Seems fine to me as-is

**Review by jeeb — APPROVED** [review]
LGTM

**Pull request merged**